guacamole-common-js/src/main/webapp/modules/Tunnel.js

```diff
@@ -907,6 +917,20 @@ Guacamole.WebSocketTunnel = function(tunnelURL) {
 
     }
 
+    /**
+     * Sends an internal "ping" instruction to the Guacamole WebSocket
+     * endpoint, verifying network connection stability. If the network is
+     * stable, the Guacamole server will receive this instruction and respond
+     * with an identical ping.
+     *
+     * @private
+     */
+    var sendPing = function sendPing() {
+        var currentTime = new Date().getTime();
+        tunnel.sendMessage(Guacamole.Tunnel.INTERNAL_DATA_OPCODE, 'ping', currentTime);
+        lastSentPing = currentTime;
+    };
+
     /**
      * Initiates a timeout which, if data is not received, causes the tunnel
```

Comment:
This comment really does not capture the totality of what's going on in this function. Would you mind adding a bit more detail here?

Reply:
Sure.